### ISSUE TITLE
Standardize GHA workflows

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/easystats#234]
# Original issue

In addition to switching to v2, maybe actions should be harmonized between repos? For example [insight](https://github.com/easystats/insight/blob/master/.github/workflows/R-check.yaml) and [datawizard](https://github.com/easystats/datawizard/blob/master/.github/workflows/R-check.yaml) have quite different configs. I wonder if it's possible to have an `easystats-actions` repo that contains a single yaml file, and then each repo calls this file.

Also, some of these repos have an action for macos-latest with R-devel, which fails when the action is not v2 (see https://github.com/easystats/insight/runs/6721433287?check_suite_focus=true for instance). Switching to v2 will get all the deps for macos-latest with R-devel the first time it is run, and this takes time (almost 3h) because binaries are [not available yet ](https://github.com/r-lib/actions/issues/546#issuecomment-1084250545). However, I think deps are cached after the first run and macos-latest with R-devel runs correctly (see a test [here](https://github.com/etiennebacher/parameters/runs/6722027784?check_suite_focus=true)).

_Originally posted by @etiennebacher in https://github.com/easystats/easystats/issues/200#issuecomment-1145698675_

---

# Update

New repository containing standardized GHA workflows for _easystats_:
https://github.com/easystats/workflows

For an example PR on how they can be used, see:
https://github.com/easystats/datawizard/pull/239

# Progress tracker:

- [x] insight 
- [x] modelbased
- [x] bayestestR
- [x] effectsize
- [x] parameters
- [x] performance
- [x] correlation
- [x] report
- [x] see
- [x] datawizard
- [x] easystats

**Comment by IndrajeetPatil** [comment]
> I wonder if it's possible to have an `easystats-actions` repo that contains a single yaml file, and then each repo calls this file.

@etiennebacher You read my mind!

Yes, this is definitely something we should work on after #200 is resolved.

**Comment by IndrajeetPatil** [comment]
Note to self: https://docs.github.com/en/actions/using-workflows/reusing-workflows

**Comment by bwiernik** [comment]
Do we maybe want to add dependabot to keep these updated? https://github.com/citation-style-language/styles/pull/6109

**Comment by etiennebacher** [comment]
Small bump for this, it's really tedious to modify all GHA one by one for each change, e.g #200, #269. I suppose there's a way to keep the global structure of the GHA and specify extra packages in each individual GHA if needed

**Comment by IndrajeetPatil** [comment]
Holy shit that worked!!!

There is now a new repository containing standardized GHA workflows for _easystats_:
https://github.com/easystats/workflows

@easystats/maintainers Please use these standardized GHA workflows in the repos you maintain.

For an example PR on how they can be used, see:
https://github.com/easystats/datawizard/pull/239

No more having to make the same changes across 10 different repositories! 😇😮‍💨

**Comment by etiennebacher** [comment]
Great news! Do you know how to deal with extra packages just for one of the repos? E.g in effectsize:

https://github.com/easystats/effectsize/blob/492928abc4577d4b56a303fc1b3dee5a1250f39b/.github/workflows/R-check.yaml#L60

**Comment by IndrajeetPatil** [comment]
I think the solution here is to conditionally skip the tests that depend on these packages, and don't let the workflows fail if the suggested packages are not available.

Currently, this might not be an issue though because we're checking only oldrel, release, and devel.

I have created a new issue to track this: https://github.com/easystats/workflows/issues/1

**Comment by etiennebacher** [comment]
Apparently you can pass inputs to the reusable workflow: https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow

That might be something to explore. But it's already great like this, thanks @IndrajeetPatil !

**Comment by IndrajeetPatil** [comment]
Thanks for looking into this. 

But, TBH, I'd prefer not to do such repo-specific customizations. We should try to make the infrastructure robust to package-to-package variations.

We can come back to this later.

**Comment by IndrajeetPatil** [comment]
Expect some GHA to fail for a while because of failure to install `XML`.

This is due to https://github.com/r-lib/actions/issues/559.
